src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -0,0 +1,1011 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Http.HPack;
+using System.Net.Security;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+// Some stuff to do:
+
+// * Handle errors/shutdown, both graceful and abortive, at connection level and per-stream
+// * Handle request cancellation
+// * Handle request body
+// * Handle frame limit and sending CONTINUATION on request headers
+// * Flow control (both inbound and outbound)
+// * Various TODOs below
+
+namespace System.Net.Http
+{
+    // TODO: This should probably implement IDisposable.
+    internal sealed class Http2Connection : HttpConnectionBase
+    {
+        private readonly SslStream _stream;
+
+        private ArrayBuffer _incomingBuffer;
+        private ArrayBuffer _outgoingBuffer;
```

Comment:
I think it'd be worth a comment on these indicating that they must not be made readonly; doing so would be disastrous :)

Reply:
Sounds good